### PR TITLE
ZQS 1104 use new paulis in time evolution

### PR DESCRIPTION
**Pull request by laurgao**
## Description

Use the new pauli operator data structures instead of openfermion. Follow-up to #41. This PR removes all usage of QubitOperator and IsingOperator in this repo outside of the `openfermion` directory.

Merge after https://github.com/zapatacomputing/orquestra-quantum/pull/44

Note that `QuantumBackendGatesTests` is not used in any orquestra repository, so I wasn't able to run tests to make sure that my changes work. (I do wonder if we need this class if it's not used anywhere?)

## Please verify that you have completed the following steps

- [x] I have self-reviewed my code.
- [x] I have included test cases validating introduced feature/fix.
- [ ] I have updated documentation.

### CODE REVIEW

File: src/orquestra/quantum/evolution.py [hunk 1]
```diff
@@ -36,20 +37,18 @@ def time_evolution(
...
 
-    terms: Iterable = list(hamiltonian.get_operators())
-
     return reduce(
         operator.add,
         (
             time_evolution_for_term(term, time / trotter_order)
             for _index_order in range(trotter_order)
-            for term in terms
+            for term in hamiltonian.terms
```
Comment (AthenaCaesura) [inline src/orquestra/quantum/evolution.py:45]: praise: This is so much more clear than the way we had it. 😄

File: src/orquestra/quantum/evolution.py [hunk 2]
```diff
@@ -62,19 +61,17 @@ def time_evolution_for_term(
 
     if len(term.terms) != 1:
```
Comment (AthenaCaesura) [inline src/orquestra/quantum/evolution.py:62]: nitpick (non-blocking): do we need this check anymore?

**Review by AthenaCaesura — APPROVED** [review]
Awesome! So much more parsble than before.

File: src/orquestra/quantum/evolution.py (hunk 2 above)
Comment (laurgao) [reply src/orquestra/quantum/evolution.py]: Fixed in [f9c6d97](https://github.com/zapatacomputing/orquestra-quantum/pull/47/commits/f9c6d9757b7201a2f3bca0c7ded7d550beadf146)

File: src/orquestra/quantum/evolution.py (hunk 1 above)
Comment (laurgao) [reply src/orquestra/quantum/evolution.py:45]: Thanks 😁

**Pull request merged**